src/lib/api/api.go

```diff
@@ -196,25 +198,54 @@ func (a *API) Query(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
+	if hash == "" {
+		switch Type {
+		case "system":
+			timeSeriesPath = "storage/system" + tsdb.FileExtension
+		case "journal":
+			timeSeriesPath = "storage/journal" + tsdb.FileExtension
+		default:
+			a.send(w, []byte("INVALID_TYPE"))
+			return
+		}
+		if ok := tsdb.VerifyChainPathExists(timeSeriesPath); !ok {
+			a.send(w, []byte("INVALID_PATH"))
+			return
+		}
+		newBlocks = []tsdb.Block{}
```

Comment:
Why is this required?

Reply:
Actually, `Query` is used for both `system` information and `tests`(ping/jitter). Incase of jitter/ping I am sending the route hash to access the in-memory chains because through `BRMatrix` we can access the chains, but we cannot access the system information like that so I am just sending the flushed data for system metrics.